Source/WebKit/UIProcess/API/APIWebAuthenticationPanelClient.h

```diff
@@ -43,9 +43,10 @@ class AuthenticatorAssertionResponse;
 
 namespace API {
 
-class WebAuthenticationPanelClient {
+class WebAuthenticationPanelClient : public RefCounted<WebAuthenticationPanelClient> {
```

Comment:
Since this is an interface, @geoffreygaren has been recommending we provide a virtual ref() / deref() in such cases and let the subclasses decide how to do their ref-counting.

Reply:
Actually that is a concrete class which is used in `WebAuthenticationPanel` as the default implementation. This is bad design but it is required like that.

Reply:
We can make it as pure abstraction class and define local class for empty implementation.

Reply:
I think we should have WebAuthenticationPanelClient subclass AbstractRefCounted. I did not request any other changes to `WebAuthenticationPanelClient`.

Reply:
> Actually that is a concrete class which is used in WebAuthenticationPanel as the default implementation. This is bad design but it is required like that.

In the future, if my review feedback is based on an incorrect assumption, please confirm with me before making further changes to avoid wasting time. I would have approved the PR as-is. Now the new PR is not as good.